### DISCUSSION TITLE
Parameter to disable header/footer/sidebar for page

### DISCUSSION THREAD

**Discussion by nikoant** [facebook/docusaurus#5046] [Q&A] [answered]
What would be the simplest way to show an existing page from Docusaurus-based website without additional layout elements like sidebar, header and footer? For the context - I'm trying to show an existing page from Docusaurus-based website in an iframe on a different website and so don't really need all these navigation items around the content. I control both websites. I can mangle CSS in the iframed page to hide its sidebar, header and footer. Are there any other ways to disable layout for page?

**Comment by slorber** [comment, marked as answer]
> What would be the simplest way to show an existing page from Docusaurus-based website without additional layout elements like sidebar, header and footer? 

If you create a file in `src/pages`, you have the full flexibility to do whatever you want, including **not using** the Layout component.

---

It's not clear to me why you want to build this page as part of your Docusaurrus site. 
Do you plan to use any docusaurus feature on this page? 
Why can't you just put a `static/myIframe.html` file and do whatever you want there? It will be served at `mySite.com/iframe`

**Reply by nikoant** [reply]
> > What would be the simplest way to show an existing page from Docusaurus-based website without additional layout elements like sidebar, header and footer?
> 
> If you create a file in `src/pages`, you have the full flexibility to do whatever you want, including **not using** the Layout component.
> 
> It's not clear to me why you want to build this page as part of your Docusaurrus site.
> Do you plan to use any docusaurus feature on this page?
> Why can't you just put a `static/myIframe.html` file and do whatever you want there? It will be served at `mySite.com/iframe`

Thanks a lot for your answer! I'll explore creating a page without layout and just including docs content there. 

Answering your question - things are a little more complicated than I described. We have a website with docs powered by Docusaurus and we have a MacOS/Windows desktop application where we want to show some of the docs in-place using webview, so users can quickly find them in context of their workflows when needed. For showing in desktop app we don't need layout so I'm looking on how to disable it.

**Reply by slorber** [reply]
So you probably have 2 choices:
- build each doc page in 2 variants, with/without layout
- build each doc once, inject custom CSS in the Webview to hide the unnecessary content

What I understand is that you don't only want to display a page in Webview, you want to display actual docs, in the ./docs folder right? 

I don't think we provide an easy way to build the docs without any layout, but that can be an interesting option. You could use the `./docs` content with 2 docs plugin instances, each with different doc plugin options (one with layout, one without for example). Building each doc twice also means you'll increase your build time.

Things to consider: 
- those docs have links, using an SPA-like navigation. How do these link should behave when run inside the webview? Can we "navigate" inside the webview or all links should open in a browser window
- what elements of the layout do you want to keep (maybe the toc?)
- is it really worth to include React? If your docs don't leverage MDX (tabs, live codeblocks, other...), plain html/CSS is enough and faster to load

**Reply by nikoant** [reply]
Yes you understood right - our aim is to have two documentation "surfaces" built from same mdx sources. One surface is our standard documentation website, and the second surface is desktop app which can show some pages from the documentation website in-place. 

I'm currently trying approach with injecting css as it looks easier, but if that wouldn't work well then we'll probably go with the second suggested solution with building two instances of each page.

- For links my idea is to try injecting a simple JS which will modify all of them with target="__blank" to open them in default browser instead of our desktop app. 
- In my tests only keeping "article" tag and all its inner content works really well for the use case.
- Plain html/css would probably work for us too as soon as images and videos can be inserted there. The pages we're going to show in-place are pretty static, but for other pages on websites we do have some React-based interactivity. The main goal here is to build docs for both surfaces from the same sources. And as we already have publishing pipeline for docusaurus docs it would be good to just re-use them for in-place docs.

**Reply by slorber** [reply]
I see, I guess it's for the Flipper desktop app 🤪 

You can definitively reuse the `./docs` folder in another tool as well. 

I actually do something a bit similar by publishing a Gatsby blog and an Expo mobile app from the same MDX folder ([blog](https://sebastienlorber.com/using-expo-in-gatsby) + [talk](https://youtu.be/ScgFQojbAAc?t=1735)), but it somehow duplicates the work as you have to manage multiple tools behaving differently. 

If Docusaurus can provide 2+ ways to use the same docs folder, it's definitively easier for you :)

I think it makes sense that we add options to build docs page as standalone pages for embedding in iframes/webview, not the first time I see this usecase. Also users previously asked for building the whole docs as a unique and long page, or create a PDF from the doc. Not exactly the same use-case but remains related.

**Comment by slorber** [comment]
Backlinking related issue: https://github.com/facebook/docusaurus/issues/7480
